src/on-create-node.ts

```diff
@@ -17,6 +17,10 @@ const onCreateNode = async (
   const query = await loadNodeContent(node)
   const queries = [ query ]
 
+  // Resolve configurations.
+  config = 'then' in config ? await config : config;
+  const { headers = {}, url, variables = {} } = config;
```

Comment:
There is no need for the reassignement here. You can `await <value>` and it will just return that value if it's not a Promise.

```diff
-  config = 'then' in config ? await config : config;
-  const { headers = {}, url, variables = {} } = config;
+  const { headers = {}, url, variables = {} } = await config;
```